freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -65,33 +94,43 @@ def handle(self, event):
             event.advisory.errata_id, True
         )
 
-        rebuild_images = []
-        auto_build_image_modules_mapping = self._get_auto_rebuild_image_mapping()
-        if auto_build_image_modules_mapping:
-            images = list(auto_build_image_modules_mapping.keys())
-            rebuild_images = self._filter_images_with_higher_rpm_nvr(images)
-        if not auto_build_image_modules_mapping or not rebuild_images:
-            msg = (
-                "Images are not enabled for auto rebuild. "
-                if not auto_build_image_modules_mapping
-                else "No images are impacted by the advisory. "
-            )
-            msg = f"{msg} message_id: {event.msg_id}"
+        try:
+            return self._handle_or_skip(event)
+        except SkipEventException as e:
+            msg = f"{e.msg} message_id: {event.msg_id}"
             db_event.transition(EventState.SKIPPED, msg)
             db.session.commit()
             self.log_info(msg)
             return []
 
+    def _handle_or_skip(self, event):
+        """Raises SkipEventException if the event should be skipped."""
+        image_modules_mapping = self._image_modules_mapping()
+        if not image_modules_mapping:
+            raise SkipEventException("No images are impacted by the advisory.")
+
+        if event.manual and event.container_images:
+            image_modules_mapping = {
+                image: modules for image, modules in image_modules_mapping.items()
+                if image in event.container_images
+            }
```

Comment:
Only published images will be present in flatpak index service, but unpublished images can be specified in request's `container_images`, especially when it's requested by release-driver where the `container_images` are always unreleased yet, however flatpak is a little special, it won't be handled by release-driver, product team is responsible for testing and releasing them, so we can ignore this at this moment.

Sometimes the requests can also be sent by person with unreleased images specified as well, I think it's not a priority to support this at this moment, I'd suggest a more meaningful skip reason like:
```
f"None of the specified images are listed in flatpak index service as latest published images impacted by the advisory: {event.container_images}"
```

Reply:
I've updated the skip reason.